c2rust-analyze/src/context.rs

```diff
@@ -665,6 +672,21 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
     pub fn iter_fns_failed(&self) -> impl Iterator<Item = DefId> + '_ {
         self.fns_failed.keys().copied()
     }
+
+    pub fn known_fn_ptr_perms<'a>(
+        &'a self,
+    ) -> impl Iterator<Item = (PointerId, PermissionSet)> + PhantomLifetime<'tcx> + 'a {
+        let tcx = &self.tcx;
+        self.fn_sigs
+            .iter()
+            .filter(|(did, _)| tcx.def_kind(tcx.parent(**did)) == DefKind::ForeignMod)
+            .filter_map(|(&did, fn_sig)| {
+                let name = tcx.item_name(did);
```

Comment:
It would be a little more robust to use `tcx.symbol_name` here, which accounts for `#[link_name]` attributes.  `extern`-block `fn`s have no substs, so it should be easy to build the `Instance` that `symbol_name` requires.

Reply:
Oh, I didn't see `tcx.symbol_name`.  That seems better.

Reply:
Changed in b608247330c051b7b7b86b9f40071080e1ed7d69.  This also changed the key of the `known_fns` map to `&'static str` since `SymbolName::name` is just a `&str`, not a `Symbol`, so the `Symbol` lookup advantage is lost if we have to intern it every time.